### DISCUSSION TITLE
re-uploading biigle report annotations and attaching them to video?

### DISCUSSION THREAD

**Discussion by CeliaHanna** [biigle/core#645] [General]
Hi there
some videos from biigle were mistakenly deleted. I have the csv file outputs for the deleted videos, as well as the videos themselves. 
Is there some way I can re upload the videos and reattach the annotations from the report?

Thanks

**Comment by mzur** [comment]
The best method is to restore the videos from a backup. Please tell me which videos should be restored and when they were deleted. You can also write info@biigle.de if the information is confidential.

**Comment by CeliaHanna** [comment]
Do you need me to grant access to the project? Coral project, dive8 files 2,3,5.

**Comment by CeliaHanna** [comment]
is it possible to re upload the videos whilst maintaining annotation information? They were deleted quite a few months ago.

**Reply by mzur** [reply]
I can restore whole volumes from the backup (including annotations). Do you know when exactly the videos were deleted? I don't want to search through all the backups.

**Reply by CeliaHanna** [reply]
Thanks! I think it was at the beginning of May 2023  but I couldn't give you an exact date, is there any other info I can give you to identify them?

**Reply by mzur** [reply]
Which project is it exactly? There is a "Coral" project but the description says "Dives 1, 2, 3, 4, 5". This project was created in June. In a backup restored from July 1st there are no DIVE8 videos in there. There is another of your projects with DIVE8 videos but it has a different name.

I need the date when the videos were deleted as exactly as possible. Otherwise I might restore a backup that does not contain all the annotations (because they were created later).

**Reply by CeliaHanna** [reply]
Hi, I apologise, the title of the project where the deleted videos are is 'Melville Bank', and the videos that were deleted are: 'DIVE8_file2', 'DIVE8_file3', and 'DIVE8_file5'. I'm sorry I cannot find an exact date for when they were deleted, it would have been in May.

**Reply by dlangenk** [reply]
Probably we (via the DB) or you can get the date of the report (you were talking about above) by going to https://biigle.de/search?q=&t=reports. If you don't have too many reports in the past you can probably find the one of the project that should be recovered. We can use the date next to it to find the correct database backup.

**Comment by CeliaHanna** [comment]
Thanks that was really helpful! I've found the latest report downloads so the most up to date annotations at the following dates: 
DIVE8_file5 : June 7 2023
DIVE8_file3 : June 6 2023
DIVE8_file2 : June 6 2023
I hope this helps. Thank you!

**Reply by mzur** [reply]
We have the videos in a backup from 2023-06-01 and in one from 2023-07-01. I've now restored the one from 2023-07-01. You can find the new volumes in your project. However, you also have to restore the video files in your S3 bucket, otherwise you can't view the annotations in the annotation tool.

**Comment by CeliaHanna** [comment]
Thank you!

I have uploaded the video again. When I click on the file, two videos show up, the old one that was deleted so is not accessible and the new one I upload. When I click on the one I uploaded I cannot see annotations, is this because I need to delete the previous video?


________________________________

**Reply by mzur** [reply]
BIIGLE is looking for the file `file2dive8jc66_1.mp4`. You have added the file `file_2_dive_8_jc66_1.mp4`. There is no need to add the file to the volume again. You only have to upload it with the correct name (`file2dive8jc66_1.mp4`) to S3 and it will work.

**Reply by mzur** [reply]
I noticed that you re-uploaded some videos but they were not displayed because BIIGLE has to process them first. I've run the processing for files 2 and 3, so these are available again. File 5 seems to be missing still.

**Comment by CeliaHanna** [comment]
Hi there, thank you I have managed to view the annotations to dive8 files 2 and 3 again :)  I have now re-uploaded dive8_file5

**Reply by CeliaHanna** [reply]
however the annotations are not showing up on the largo tool. Is this expected?

**Reply by mzur** [reply]
Video 5 is now reprocessed and available again. The Largo thumbnails are missing because the video files were not available when I imported the volumes. They will now be reprocessed, too, and should appear shortly.
